### PR TITLE
refac: refactor verifier part 1

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR prepares for refactoring of the verifier. The verifier is planned to be refactored as we did in #294, so we plan to create `plonk::PermutationVerifier`, `lookup::Verifier` and `plonk::VanishingVerifier`.

### CODE REVIEW

File: tachyon/zk/expressions/evaluator/proving_evaluator.h [hunk 1]
```diff
@@ -27,14 +27,14 @@
 namespace tachyon::zk {
 
 template <typename Evals>
-class SimpleEvaluator
+class ProvingEvaluator
```
Comment (dongchangYoo) [inline tachyon/zk/expressions/evaluator/proving_evaluator.h:30]: how about "LookupEvaluator"?
That's why, It is used for only Lookup compression.

**Review by dongchangYoo — CHANGES_REQUESTED** [review]

File: tachyon/zk/plonk/permutation/permutation_argument.h [hunk 2]
```diff
@@ -35,36 +35,36 @@ class TACHYON_EXPORT PermutationArgument {
...
-    // degree 2:
-    // l_first(X) * (1 - z(X)) = 0
+    // See https://zcash.github.io/halo2/design/proving-system/permutation.html
+    // for more details.
     //
-    // We will fit as many polynomials pᵢ(X) as possible
-    // into the required degree of the circuit, so the
-    // following will not affect the required degree of
-    // this middleware.
+    // degree 2: l_first(X) * (1 - Z(X)) = 0
```
Comment (ashjeong) [inline tachyon/zk/plonk/permutation/permutation_argument.h:41]: Either put the operations on the next row, or change every other comment regarding `degree [#]`to also be in-line

File: tachyon/zk/plonk/permutation/permutation_argument.h [hunk 3]
```diff
@@ -35,36 +35,36 @@ class TACHYON_EXPORT PermutationArgument {
...
+    // degree of the circuit, so the following will not affect the required
+    // degree of this middleware.
     //
     // clang-format off
-    // (1 - (l_last(X) + l_blind(X))) * (z(ω * X) Π (p(X) + β * sᵢ(X) + γ) - z(X) Π (p(X) + β * δⁱ * X + γ))
+    // (1 - (l_last(X) + l_blind(X))) * (Z(ω * X) Π (pᵢ(X) + β * sᵢ(X) + γ) - Z(X) Π (pᵢ(X) + β * δⁱ * X + γ))
     // clang-format on
     //
     // On the first sets of columns, except the first
     // set, we will do
```
Comment (ashjeong) [inline tachyon/zk/plonk/permutation/permutation_argument.h:52]:
```suggestion
    // On the first sets of columns, except the first set, we will do
```

File: tachyon/zk/plonk/constraint_system/constraint_system.h [hunk 4]
```diff
@@ -518,6 +518,8 @@ class ConstraintSystem {
     return *cached_blinding_factors_;
   }
 
+  RowOffset ComputeLastLow() const { return -(ComputeBlindingFactors() + 1); }
```
Comment (ashjeong) [inline tachyon/zk/plonk/constraint_system/constraint_system.h:521]:
```suggestion
  RowOffset ComputeLastRow() const { return -(ComputeBlindingFactors() + 1); }

**Review by ashjeong — COMMENTED** [review]
https://github.com/kroma-network/tachyon/pull/324/commits/3b31ce88453c4cf1c7c8f19fcae7d46c3f94bb89
“use a proper namespace” -> “use the proper namespace”

https://github.com/kroma-network/tachyon/pull/324/commits/f6643bcab0888ed2b82277e0399c1a09eb8248dc
“In addition, this also sort the methods in the order poly, evals and container.” -> “In addition, this also sorts the methods in the order of poly, evals and containers."

File: tachyon/zk/expressions/evaluator/proving_evaluator.h (hunk 1 above)
Comment (chokobole) [reply tachyon/zk/expressions/evaluator/proving_evaluator.h:30]: Good point

File: tachyon/zk/plonk/constraint_system/constraint_system.h (hunk 4 above)
Comment (chokobole) [reply tachyon/zk/plonk/constraint_system/constraint_system.h:521]: OMG..

**Review by Insun35 — APPROVED** [review]
LGTM

File: tachyon/zk/lookup/halo2/prover_impl.h [hunk 5]
```diff
@@ -239,10 +245,15 @@ void Prover<Poly, Evals>::Open(
     const LookupPair<BlindedPolynomial<Poly, Evals>>& permuted_pair =
         permuted_pairs_[i];
 
+    // Zₗ,ᵢ(x)
```
[the remark this answers is not in the document]
Comment (chokobole) [reply tachyon/zk/lookup/halo2/prover_impl.h:248]: No subscription on `L` and `p` is a small capital!

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Review by ashjeong — APPROVED** [review]
LGTM

**Pull request merged**